Python/ast.c

```diff
@@ -4265,6 +4266,12 @@ decode_bytes_with_escapes(struct compiling *c, const node *n, const char *s,
     return result;
 }
 
+/* Forward declarations for recursive functions. */
```

Comment:
Yet a note. I don't see a recursion except a self-recursion in `fstring_shift_node_locations()`. If change the order of function definitions, forward declarations will become not needed.